src/common/pick_address.cc

```diff
@@ -164,27 +164,34 @@ void pick_addresses(CephContext *cct, int needs)
     exit(1);
   }
 
+  auto& public_addr = cct->_conf->get_val<entity_addr_t>("public_addr");
```

Comment:
could you be more consistent with using `auto`? after a second thought, i think use `&` could be dangerous: the `get_val()` returns a reference to its caller, what if the stored value is changed by another thread? there is a racing here. i am trying to fix this in #18934